lib/sycamore/sycamore/utils/time_trace.py

```diff
@@ -95,3 +135,45 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+class _ZeroRU:
+    def __init__(self):
+        self.ru_utime = 0
+        self.ru_stime = 0
+
+
+class LogTime:
+    def __init__(self, name, point: bool = False, log_start: bool = False):
+        if point:
+            self._logpoint(name)
+            return
+
+        self.name = name
+        self.log_start = log_start
+
+    def __enter__(self):
+        self.start()
+        if self.log_start:
+            self._logpoint(self.name + "_start")
```

Comment:
Any reason not to have this code in start()?

Reply:
I had thought that log_start outside of context didn't make sense, but changed.